spec/requests/preference/preference_spec.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'Preferences' do
+  describe 'GET index' do
+    subject { get preferences_path }
+
+    context 'when logged in' do
+      let!(:user) { create(:user) }
+      let!(:preference) { create(:preference, user: user) }
+
+      before { sign_in user }
+
+      it 'has http status 200' do
+        expect(subject).to eq(200)
+      end
+
+      it 'assigns user preferences' do
+        subject
+        expect(assigns(:preferences)).to eq([preference])
+      end
+    end
+  end
+
+  describe 'GET new' do
+    subject { get new_preference_path }
+
+    context 'when logged in' do
+      let!(:user) { create(:user) }
+
+      before { sign_in user }
+
+      it 'has http status 200' do
+        expect(subject).to eq(200)
+      end
+
+      it 'assigns a new preference' do
+        subject
+        expect(assigns(:preference)).to be_a_new(Preference)
+      end
```

Comment:
Comment: assigns was deprecated ([here](https://apidock.com/rails/ActionController/TestProcess/assigns)) on latest version of rails so it is better not to use it.

I think that maybe the best way to test this (the :new Form page) would be with feature tests. As we discussed it is not the priority now but I will leave you an [article](https://www.codewithjason.com/rails-integration-tests-rspec-capybara/) about different types of testing with Rspec in case you are interested on the future.